app/src/main/java/org/sagebionetworks/research/mpower/tracking/recycler_view/TriggersLoggingItemViewHolder.java

```diff
@@ -0,0 +1,72 @@
+package org.sagebionetworks.research.mpower.tracking.recycler_view;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.Transformations;
+import android.support.annotation.NonNull;
+import android.support.v7.widget.RecyclerView.ViewHolder;
+
+import org.sagebionetworks.research.mpower.tracking.fragment.TriggersLoggingFragment;
+import org.sagebionetworks.research.mpower.tracking.model.TrackingItem;
+import org.sagebionetworks.research.mpower.tracking.view_model.configs.SimpleTrackingItemConfig;
+import org.sagebionetworks.research.mpower.tracking.view_model.logs.SimpleTrackingItemLog;
+import org.sagebionetworks.research.mpower.tracking.view_model.TrackingTaskViewModel;
+import org.sagebionetworks.research.mpower.tracking.widget.TriggersLoggingUIFormItemWidget;
+import org.threeten.bp.Instant;
+
+/**
+ * View Holder for the Logging Items from the Triggers Task.
+ */
+public class TriggersLoggingItemViewHolder extends ViewHolder {
+    private TriggersLoggingUIFormItemWidget widget;
+    private TrackingTaskViewModel<SimpleTrackingItemConfig, SimpleTrackingItemLog> viewModel;
+    private SimpleTrackingItemConfig config;
+    private TriggersLoggingFragment triggersLoggingFragment;
+
+    public TriggersLoggingItemViewHolder(final TriggersLoggingUIFormItemWidget itemView,
+            final TrackingTaskViewModel<SimpleTrackingItemConfig, SimpleTrackingItemLog> viewModel,
+            final TriggersLoggingFragment triggersLoggingFragment) {
+        super(itemView);
+        this.widget = itemView;
+        this.viewModel = viewModel;
+        this.triggersLoggingFragment = triggersLoggingFragment;
+    }
+
+    public void setContent(@NonNull SimpleTrackingItemConfig config) {
+        this.config = config;
+        boolean logged = this.viewModel.isLogged(config);
+        this.widget.setLogged(logged);
+        this.widget.getTitle().setText(config.getTrackingItem().getIdentifier());
+        this.setRecordButtonListener();
+        this.setUndoButtonListener();
+        this.setLogObservers(this.config.getTrackingItem());
+    }
+
+    private void setLogObservers(@NonNull TrackingItem trackingItem) {
+        LiveData<SimpleTrackingItemLog> logLiveData = Transformations.map(this.viewModel.getLoggedElements(), elements -> {
+            for (SimpleTrackingItemLog log : elements) {
```

Comment:
Seems this Transformation is used multiple times and could be refactored

Reply:
Or perhaps the ViewModel could do a Map<Element,LiveData<Log>> so it's computed once (or LiveData<Map<Element<LiveData<Log>>>

pretty nit-picky at this point

Reply:
In a future pr I'm planning on making the log/config retrieval use some sort of a map.